frontend/src/store/flowSlice.ts

```diff
@@ -735,9 +827,17 @@ const flowSlice = createSlice({
 
       // Update the node title
       const node = state.nodes.find(node => node.id === nodeId);
-      if (node && node.data) {
-        node.data.config = {
-          ...node.data.config,
+      if (node) {
+        node.data = {
+          ...node.data,
+          title: newTitle
+        };
+      }
+
+      // Update nodeConfigs with the new title
+      if (state.nodeConfigs[nodeId]) {
+        state.nodeConfigs[nodeId] = {
+          ...state.nodeConfigs[nodeId],
           title: newTitle
         };
       }
```

Comment:
In the handling of the downstream nodes below, `node.data.config` is referenced again. Is that correct? Since `config` is now stored `state.nodeConfigs` object.